runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -4788,341 +4800,293 @@ J9::Z::TreeEvaluator::anewArrayEvaluator(TR::Node * node, TR::CodeGenerator * cg
       return TR::TreeEvaluator::VMnewEvaluator(node, cg);
    }
 
-///////////////////////////////////////////////////////////////////////////////////////
-// multianewArrayEvaluator:  multi-dimensional new array of objects
-// NB Must only be used for arrays of at least two dimensions
-///////////////////////////////////////////////////////////////////////////////////////
-static TR::Register * generateMultianewArrayWithInlineAllocators(TR::Node *node, TR::CodeGenerator *cg)
-   {
+/**
+ * \brief
+ * Inline allocation of two dimensional new array.
+ *
+ * \details
+ * Allocate 2 dimensional multi new arrays of any type with inline instructions if the size is within the range of TLH.
+ * This helper needs the following conditions to work properly and the caller should check these before calling:
+ * 1. Must be a 64bit system.
+ * 2. CPU version S390 Z196 and higher.
+ * 3. Number of dimensions is 2.
+ * 4. Component size is a valid type size.
+ * This helper allocate any non negative size 2D array if it fits inside the TLH.
+ *
+ * \param node
+ * The node.
+ *
+ * \param cg
+ * The code generator.
+ *
+ * \param componentSize
+ * The leaf array component size.
+ *
+ * \return
+ * TR::Register with the address of the new object.
+ */
+static TR::Register * generateMultianewArrayWithInlineAllocators(TR::Node *node, TR::CodeGenerator *cg, int32_t componentSize)
+{
    #define iComment(str) if (compDebug) compDebug->addInstructionComment(cursor, (const_cast<char*>(str)));
    TR::Compilation *comp = cg->comp();
    TR_Debug *compDebug = comp->getDebug();
-   TR_ASSERT_FATAL(comp->target().is64Bit(), "multianewArrayEvaluator is only supported on 64-bit JVMs!");
```

Comment:
Did we remove this ASSERT ? or it is just placed differently. - If it is removed, do you claim to support this on 31-bit JVM?